clwb/src/com/google/idea/blaze/clwb/run/BazelDebugFlagsBuilder.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2025 The Bazel Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.idea.blaze.clwb.run
+
+import com.google.common.collect.ImmutableList
+import com.intellij.openapi.util.registry.Registry
+import com.intellij.util.system.OS
+import com.jetbrains.cidr.lang.workspace.compiler.*
+
+/**
+ * Builds flags for debugging a blaze target, flags are either used for just
+ * building a binary [BlazeCidrRunConfigurationRunner] or for running actually
+ * running the binary [BlazeGDBServerProvider].
+ *
+ * In theory we would like to have one builder that builds a BlazeCommand to
+ * have full control over the environment in the builder.
+ */
+class BazelDebugFlagsBuilder(
+  private val debuggerKind: BlazeDebuggerKind,
+  private val compilerKind: OCCompilerKind,
```

Comment:
some compilerkind/debuggerkind combinations do not make any sense, do we want to enforce some checks here or around?